sys/kern/kern_exec.c

```diff
@@ -1306,10 +1307,17 @@ exec_map_stack(struct image_params *imgp)
 		 * We currently place it just just below the stack as
 		 * this avoides collisions with init which is linked
 		 * near the bottom of the address space.
+		 *
+		 * The extra page beyond ARG_MAX accounts for space
+		 * needed for the ELF auxiliary, argument, and
+		 * environment vectors and other data stored in this
+		 * space other than argument and environment strings.
 		 */
+		strings_size = CHERI_REPRESENTABLE_LENGTH(ARG_MAX + PAGE_SIZE);
```

Comment:
We could perhaps compute a more exact size here.  imgp->auxargs is currently initialized after exec_map_stack but that could be fixed in imgact_elf.c.  The rest of the data is available such that we could call a helper function here to compute the true size, but it would need to probably by a new sv_* hook that mirrors exec_copyout_strings somewhat, but only for SV_CHERI.